02-express-tutorial/app.js

```diff
@@ -1,6 +1,60 @@
 const { products } = require("./data");
 const express = require("express");
 const app = express();
+const peopleRouter = require('./routes/people');
+
+// app.use(express.static('./methods-public'));
+
+app.use(express.urlencoded({ extended: false }));
+app.use(express.json());
+app.use('/api/v1/people', peopleRouter);
+
+// app.get('/api/v1/people', (req, res) => {
+//     res.status(200).json({ success: true, data: people });
+// });
+
+// app.post('/api/v1/people', (req, res) => {
+//     const { name } = req.body;
+//     if(!name){
+//         return res.status(400).json({ success: false, message: 'Please provide a name' })
+//     }
+//     people.push({ id: people.length + 1, name: req.body.name });
+//     res.status(201).json({ success: true, name: req.body.name })
+// });
+
+// app.put('/api/v1/people/:id', (req, res) => {
+//     const { id } = req.params;
+//     const { name } = req.body;
+
+//     const person = people.find((person) => person.id === Number(id))
+//     if(!person){
+//         return res.status(400).json({ success: false, message: `No person with id ${id}` })
+//     }
+//     const newPeople = people.map((person) => {
+//         if(person.id === Number(id)){
+//             person.name = name
+//         }
+//         return person
+//     });
+//     res.status(200).json({success: true, data: newPeople })
+// });
+
+// app.delete('/api/v1/people/:id', (req, res) => {
+//     const person = people.find((person) => person.id === Number(req.params.id))
+//     if(!person){
+//         return res.status(400).json({ success: false, message: `No person with id ${req.params.id}` })
+//     }
+//     const newPeople = people.filter((person) => person.id !== Number(req.params.id));
+
+const logger = (req, res, next) => {
```

Comment:
This is a good job.. But u need to put this logger function above your routes (e.g. line 10) so that the logger would capture the log before the route is reached. Otherwise, majority of requests would not be logged, because they would not get to the stage of that request before getting resolved.

Reply:
Thank you for the feedback! I'll fix it to ensure the logger works properly.